rskj-core/src/main/java/org/ethereum/rpc/dto/BlockResultDTO.java

```diff
@@ -0,0 +1,259 @@
+import org.ethereum.core.BlockHeader;
+import org.ethereum.core.Transaction;
+import org.ethereum.db.BlockStore;
+import org.ethereum.rpc.TypeConverter;
+
+import java.util.*;
+
+import static org.ethereum.rpc.TypeConverter.toJsonHex;
+
+public class BlockResultDTO {
+    private final String number; // QUANTITY - the block number. null when its pending block.
+    private final String hash; // DATA, 32 Bytes - hash of the block. null when its pending block.
+    private final String parentHash; // DATA, 32 Bytes - hash of the parent block.
+    private final String sha3Uncles; // DATA, 32 Bytes - SHA3 of the uncles data in the block.
+    private final String logsBloom; // DATA, 256 Bytes - the bloom filter for the logs of the block. null when its pending block.
+    private final String transactionsRoot; // DATA, 32 Bytes - the root of the transaction trie of the block.
+    private final String stateRoot; // DATA, 32 Bytes - the root of the final state trie of the block.
+    private final String receiptsRoot; // DATA, 32 Bytes - the root of the receipts trie of the block.
+    private final String miner; // DATA, 20 Bytes - the address of the beneficiary to whom the mining rewards were given.
+    private final String difficulty; // QUANTITY - integer of the difficulty for this block.
+    private final String totalDifficulty; // QUANTITY - integer of the total difficulty of the chain until this block.
+    private final String extraData; // DATA - the "extra data" field of this block
+    private final String size;//QUANTITY - integer the size of this block in bytes.
+    private final String gasLimit;//: QUANTITY - the maximum gas allowed in this block.
+    private final String gasUsed; // QUANTITY - the total used gas by all transactions in this block.
+    private final String timestamp; //: QUANTITY - the unix timestamp for when the block was collated.
+    private final List<Object> transactions; //: Array - Array of transaction objects, or 32 Bytes transaction hashes depending on the last given parameter.
+    private final List<String> uncles; //: Array - Array of uncle hashes.
+    private final String minimumGasPrice;
+    private final String bitcoinMergedMiningHeader;
+    private final String bitcoinMergedMiningCoinbaseTransaction;
+    private final String bitcoinMergedMiningMerkleProof;
+    private final String hashForMergedMining;
+
+    public BlockResultDTO(
+            Long number,
+            Keccak256 hash,
+            Keccak256 parentHash,
+            byte[] sha3Uncles,
+            byte[] logsBloom,
+            byte[] transactionsRoot,
+            byte[] stateRoot,
+            byte[] receiptsRoot,
+            RskAddress miner,
+            BlockDifficulty difficulty,
+            BlockDifficulty totalDifficulty,
+            byte[] extraData,
+            int size,
+            byte[] gasLimit,
+            long gasUsed,
+            long timestamp,
+            Object[] transactions,
```

Comment:
This PR changes the internal representation from `[]` to `List`. 
I suggest using `List` also at the constructor to avoid unnecessary conversions.